File: new-resource-detector/src/test/java/nl/haploid/octowight/service/ResourceProducerServiceIT.java
package nl.haploid.octowight.service;

import nl.haploid.octowight.TestData;
import nl.haploid.octowight.AbstractIT;
import org.apache.kafka.clients.producer.RecordMetadata;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.concurrent.Future;

import static org.junit.Assert.assertNotNull;

public class ResourceProducerServiceIT extends AbstractIT {

	@Autowired
	private ResourceProducerService service;

	@Test
	public void testPublishResourceDescriptor() throws Exception {
		final ResourceDescriptor descriptor = TestData.resourceDescriptor(555l);
		final Future<RecordMetadata> future = service.publishResourceDescriptor(descriptor);
		assertNotNull(future.get());
	}
}

File: sample-resource-builder/src/test/scala/nl/haploid/octowight/sample/service/ResourceProducerServiceIT.scala
package nl.haploid.octowight.sample.service

import nl.haploid.octowight.JsonMapper
import nl.haploid.octowight.kafka.KafkaConsumerFactory
import nl.haploid.octowight.registry.data.ResourceMessage
import nl.haploid.octowight.sample.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.{Autowired, Value}

class ResourceProducerServiceIT extends AbstractIT {
  @Autowired private[this] val resourceProducerService: ResourceProducerService = null
  @Autowired private[this] val kafkaConsumerFactory: KafkaConsumerFactory = null
  @Autowired private[this] val jsonMapper: JsonMapper = null
  @Value("${octowight.kafka.topic.resources.built}") private[this] val topic: String = null

  behavior of "Resource producer service"

  it should "send resources" in {
    val resource1 = TestData.resource
    val resource2 = TestData.resource
    val expectedResources: Iterable[ResourceMessage] = Iterable(resource1, resource2)
    resourceProducerService.send(expectedResources)
    val kafkaConsumer = kafkaConsumerFactory.kafkaConsumer(topic)
    val actualResources = kafkaConsumer.nextMessages(3)
      .map(jsonMapper.deserialize(_, classOf[ResourceMessage]))
    actualResources should be(expectedResources)
  }
}
